source/util/StringUtils.cpp

```diff
@@ -17,7 +17,7 @@ namespace Aws
             namespace Util
             {
                 // maximum buffer size after which log messages will get truncated
-                static const int MAX_BUF_SIZE = 4096;
+                static const u_int16_t MAX_BUF_SIZE = 5000;
```

Comment:
`u_int16_t` is windows-specific, use `std::uint16_t` (from `<cstdint>` header) for cross-platform code. For this use case I recommend using `std::size_t` (from `<cstddef>`)  to match the type of `vsnprintf`.

https://en.cppreference.com/w/cpp/header/cstdint